plugins/processors/t128_transform/t128_transform_state_change_test.go

```diff
@@ -74,182 +74,254 @@ func TestStateChangeLeavesUnmarkedFieldsInTact(t *testing.T) {
+			Fields: map[string]string{"/state": "/non-state"},
+			Samples: []sample{
+				{"/non-state": 45},
+				{"/non-state": 50},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: true,
+			Result:         sample{"/state": int64(50)},
+		},
+		{
+			Name:   "remove-mismatching-expired",
+			Fields: map[string]string{"/state": "/non-state"},
+			Samples: []sample{
+				{"/non-state": 45},
+				{"/non-state": 50},
+			},
+			TimeDelta:      6 * time.Second,
+			RemoveOriginal: true,
+			Result:         sample{"/state": int64(50)},
+		},
+		{
+			Name:   "leave-matching-new",
+			Fields: map[string]string{"/state": "/state"},
+			Samples: []sample{
+				{},
+				{"/state": 50},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/state": int64(50)},
+		},
+		{
+			Name:   "leave-matching-existing-same",
+			Fields: map[string]string{"/state": "/state"},
+			Samples: []sample{
+				{"/state": 45},
+				{"/state": 45},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{},
+		},
+		{
+			Name:   "leave-matching-existing-changed",
+			Fields: map[string]string{"/state": "/state"},
+			Samples: []sample{
+				{"/state": 45},
+				{"/state": 50},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/state": int64(50)},
+		},
+		{
+			Name:   "leave-matching-expired",
+			Fields: map[string]string{"/state": "/state"},
+			Samples: []sample{
+				{"/state": 45},
+				{"/state": 50},
+			},
+			TimeDelta:      6 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/state": int64(50)},
+		},
+		{
+			Name:   "leave-mismatching-new",
+			Fields: map[string]string{"/state": "/non-state"},
+			Samples: []sample{
+				{},
+				{"/non-state": 50},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/non-state": int64(50), "/state": int64(50)},
+		},
+		{
+			Name:   "leave-mismatching-existing-same",
+			Fields: map[string]string{"/state": "/non-state"},
+			Samples: []sample{
+				{"/non-state": 45},
+				{"/non-state": 45},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/non-state": int64(45)},
+		},
+		{
+			Name:   "leave-mismatching-existing-changed",
+			Fields: map[string]string{"/state": "/non-state"},
+			Samples: []sample{
+				{"/non-state": 45},
+				{"/non-state": 50},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/non-state": int64(50), "/state": int64(50)},
+		},
+		{
+			Name:   "leave-mismatching-expired",
+			Fields: map[string]string{"/state": "/non-state"},
+			Samples: []sample{
+				{"/non-state": 45},
+				{"/non-state": 50},
+			},
+			TimeDelta:      6 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/non-state": int64(50), "/state": int64(50)},
+		},
+
+		{
```

Comment:
This is where new tests start. The earlier changes are just converting to `Samples` and `Result` to make the test more flexible.